Makefile

```diff
@@ -1,8 +1,8 @@
 include common.mk
 
-CLI_DIR:=$(realpath $(CURDIR)/../cli)
-ENGINE_DIR:=$(realpath $(CURDIR)/../engine)
-STATIC_VERSION:=$(shell static/gen-static-ver $(ENGINE_DIR) $(VERSION))
+CLI_DIR=$(realpath $(CURDIR)/../cli)
```

Comment:
What about `?=`  so user can override each directory independently ?

Reply:
Makefile variables / macros are confusing; I _think_ that's already possible even with the current change in this PR, but I need to double-check, because these variables always confuse me.

Reply:
Looks like that works;

Added a target for debugging:

```patch
diff --git a/Makefile b/Makefile
index 3a8142b..4bcaa7b 100644
--- a/Makefile
+++ b/Makefile
@@ -35,3 +35,7 @@ static: ## build static-compiled packages
        for p in $(DOCKER_BUILD_PKGS); do \
                $(MAKE) -C $@ VERSION=$(VERSION) ENGINE_DIR=$(ENGINE_DIR) CLI_DIR=$(CLI_DIR) GO_VERSION=$(GO_VERSION) $${p}; \
        done
+
+.PHONY: foo
+foo:
+       @echo CLI_DIR is $(CLI_DIR) !!
```


Created a Makefile.test:

```makefile
CLI_DIR=/hello-world

foo:
	@make foo

bar:
	@make CLI_DIR=$(CLI_DIR) foo
```

Tried some variations:

```bash
make -f Makefile.test CLI_DIR=override foo
CLI_DIR is override !!

make -f Makefile.test foo
CLI_DIR is /Users/sebastiaan/go/src/github.com/docker/cli !!

make -f Makefile.test bar
CLI_DIR is /hello-world !!
```